skyportal/models.py

```diff
@@ -637,11 +637,17 @@ def token_groups(self):
 Token.groups = token_groups
 
 
+def delete_obj_if_all_data_owned(cls, user_or_token):
```

Comment:
Shouldn't the check re: associated data happen here rather than via a joinedload passed to the query method?

Reply:
That was my first thought as well, but the joinedload method was much simpler to implement so I went with that instead. Doing that logic here would involve doing a bunch of `query_rows_accessible_by` queries for all the associated data, joining them against the Obj, etc. to look for Objs that are allowed to be deleted. The way I have it right now, I just let the loop in the `verify_and_commit` check all those for me instead of having it happen when I first fetch the Obj to be deleted. That said, I agree that doing the check here would be the more correct way of doing it from a good-coding-practices standpoint, so I'm happy to make that change if you feel I should.

Reply:
Yeah, sorry to request a revision here, but I do think the access policy defined here should do all of the relevant checks, just to be consistent with how they're implemented and used everywhere else. That way, if we ever need to call `Obj.get_if_accessible_by(..., mode="delete")` anywhere else, we can rest assured the correct checks are still happening. I think it should be abstracted away here.

Reply:
No problem, that makes perfect sense. I'll work on that